lib/puppetx/simp/compliance_mapper.rb

```diff
@@ -152,9 +138,8 @@ def initialize(object)
     def load(options={}, &block)
       @callback.debug("callback = #{callback.codebase}")
 
-      module_scope_compliance_map = callback.cached_lookup "compliance_markup::compliance_map", {}, &block
-      top_scope_compliance_map    = callback.cached_lookup "compliance_map", {}, &block
-
+      module_scope_compliance_map = block.call 'compliance_markup::compliance_map', {}
```

Comment:
Can we go ahead and verify that this is actually doing what we think it's doing or pass `context` into the method instead?

Reply:
> Can we go ahead and verify that this is actually doing what we think it's doing or pass `context` into the method instead?

https://github.com/simp/pupmod-simp-compliance_markup/blob/bbdb7d41928f4f5c9b7d557a81ab58b433c0eb85/lib/puppet/functions/compliance_markup/enforcement.rb#L81 is the source of the `lookup`, and the test for `compliance_markup::compliance_map` is working.

Reply:
But where does it come from in the calling code? I get that it's what it happens to be but passing a block down that many calls means that a seemingly innocent update later on might result in a very difficult to debug problem.

Reply:
I guess what I'm getting at is that we either need to add documentation in the code on both sides of the block to make sure we know what we might affect or just change it to passing `context` so everything is explicit.

Reply:
The comment below from @kendall-moore looks like a placeholder for figuring out how this works so that's fine for now.

Could someone please update that with the relevant details and/or a link back to this conversation for the future?

I'll leave this unresolved so that there is context present.